### PR TITLE
[lab-random-forests-Rita]

### PR DESCRIPTION
**Pull request by ritasilv**

### CODE REVIEW

**Comment by AnaAGG** [comment]
# lab random forest

Hi Rita 🙋🏻‍♀️! We continue with more new machine learning models

Ok Rita, the workflow of your lab is perfect, buuut I have a detail to tell you. You initialized the model with a given parameters, but are these parameter the best option to fit our model? i.e. These parameters give us the best accuracy? 

However, in python we have the method `GridSearchCV` methos to find the best hyperparameter combination of our model. 


[Here](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html) somo documentacion about `GridSearchCV` method. 


How can we do it?

```python
# set the parameters we want to check. The decission, sometimes is random 😔 or based in our previous knowledge. 
param_grid = {
    'max_depth' = [2, 3, 4, 5]
    'n_estimators': [50, 100],
    'min_samples_split': [2, 4],
    'min_samples_leaf' : [1, 2],
    'max_features': ['sqrt']
    }

# initialize the model
clf = RandomForestClassifier(random_state=100) 

# execute the pipeline created before
grid_search = GridSearchCV(clf, param_grid, cv=5,return_train_score=True,n_jobs=-1)

# fit the model
grid_search.fit(X_train,y_train)

# git the `best_params_` method we can get the best model of our pipeline
grid_search.best_params_


# once we have the best combination of parameters we initialize the model with those parameters

clf = RandomForestClassifier(random_state=0, max_features='sqrt', min_samples_leaf=1, min_samples_split=4, n_estimators=100)

# Finally, calculate the accuracy of our model
cross_val_scores = cross_val_score(clf, X_train, y_train, cv=10)
print(np.mean(cross_val_scores))

```

All in all good job in this lab Rita!  Go go go, we are in the home stretch 💪!


image
